File: app/src/main/java/ru/michanic/mymot/Models/SectionModelItem.java
package ru.michanic.mymot.Models;

public class SectionModelItem {

    String sectionTitle = null;
    Model model = null;

    public SectionModelItem(String sectionTitle) {
        this.sectionTitle = sectionTitle;
    }

    public SectionModelItem(Model model) {
        this.model = model;
    }

    public void setSectionTitle(String sectionTitle) {
        this.sectionTitle = sectionTitle;
    }

    public String getSectionTitle() {
        return sectionTitle;
    }


    public void setModel(Model model) {
        this.model = model;
    }

    public Model getModel() {
        return model;
    }
}

File: app/src/main/java/ru/michanic/mymot/Kotlin/Models/SectionModelItem.kt
package ru.michanic.mymot.Kotlin.Models

class SectionModelItem {
    var sectionTitle: String? = null
    var model: Model? = null
    var propertyTitle: String? = null
    var propertyValue: String? = null
    val type: Int

    constructor(sectionTitle: String?) {
        type = SECTION_TITLE
        this.sectionTitle = sectionTitle
    }

    constructor(model: Model?) {
        type = LIST_MODEL
        this.model = model
    }

    constructor(propertyTitle: String?, propertyValue: String?) {
        this.propertyTitle = propertyTitle
        if (propertyValue != null) {
            type = PRICE_CELL
            this.propertyValue = propertyValue
        } else {
            type = SIMPLE_CELL
        }
    }

    companion object {
        const val SECTION_TITLE = 0
        const val LIST_MODEL = 1
        const val SIMPLE_CELL = 2
        const val PRICE_CELL = 3
        const val PRICE_FROM_NAME = "От"
        const val PRICE_FOR_NAME = "До"
    }
}